src/components/features/actionsMenu/VendorTableActionsMenu.tsx

```diff
@@ -3,34 +3,100 @@ import Menu from '@mui/material/Menu'
 import MenuItem from '@mui/material/MenuItem'
 import MoreHorizIcon from '@mui/icons-material/MoreHoriz'
 import React from 'react'
+import WarningActionDialog from '../../shared/WarningActionDialog'
 import { useTranslation } from 'react-i18next'
-import WarningActionDialog from '@/components/shared/WarningActionDialog'
+import { VendorDto, VendorFormDto } from '@/services/model'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { getWarehouseManagementApi } from '@/services/generated-api'
+import { useSnackbar } from '@/hooks/useSnackbar'
+import FormDialog from '../../shared/FormDialog'
+import { NewVendorFormData, newVendorSchema } from '@/schemas/newVendorSchema'
+import NewVendorForm from '../forms/NewVendorForm'
+import { BodyType } from '@/services/api'
+import { SubmitHandler } from 'react-hook-form'
 
-interface VendorTableActionsMenuProps {
-  specificOptionHandler: Function
-  options: { title: string; value: string }[]
+interface VendorsTableActionsMenuProps {
+  vendor: VendorDto
 }
 
-export default function VendorTableActionsMenu({
-  specificOptionHandler,
-  options
-}: TableActionsMenuProps) {
+export default function VendorTableActionsMenu({ vendor }: VendorsTableActionsMenuProps) {
   const { t: translate } = useTranslation()
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
   const open = Boolean(anchorEl)
+  const [selectedOption, setSelectedOption] = React.useState<string | null>(null)
+
+  const { showSnackbar } = useSnackbar()
 
   const handleClick = (event: React.MouseEvent<HTMLElement>) => {
     setAnchorEl(event.currentTarget)
   }
   const handleClose = () => {
+    setSelectedOption(null)
     setAnchorEl(null)
   }
 
-  const actionHandler = (option: string) => {
-    specificOptionHandler(option)
+  const onDiscardClick = () => {
     handleClose()
   }
 
+  const queryClient = useQueryClient()
+
+  const mutationDelete = useMutation({
```

Comment:
you can check out what @Hristo-Boyadzhiev has done with the zones in `hooks/services/zones` and you can move the query hooks and make them like what he has done to make things simpler and easier to read.

Reply:
It is fixed now, also i call TableActionsMenu to build dynamicly buttons for actions menu